aequilibrae/paths/graph.py

```diff
@@ -249,6 +249,13 @@ def _build_directed_graph(self, network: pd.DataFrame, centroids: np.ndarray):
         present_centroids = np.isin(centroids, nodes, assume_unique=True)
         if not present_centroids.all():
             warnings.warn("Found centroids not present in the graph!\n" + str(centroids[~present_centroids]))
+                        missing_centroids = centroids[~np.isin(centroids, nodes)]
+            warnings.warn(f"Found {len(missing_centroids)} centroids not present in the graph!")
+            # Save to disk for network debugging and fixing
+            output_path = os.path.join(os.getcwd(), "missing_centroids_from_graph.csv")
+            pd.Series(missing_centroids).to_csv(output_path, index=False, header=False)
+            print(f"Missing centroids written to: {output_path}")
+            
```

Comment:
This looks beneficial, but there is already a warning logged which this duplicates. 

```suggestion

            missing_centroids = centroids[~np.isin(centroids, nodes)]
            num_missing = len(missing_centroids)
            
            # Save to disk for network debugging and fixing
            output_path = os.path.join(os.getcwd(), "missing_centroids_from_graph.csv")
            pd.Series(missing_centroids).to_csv(output_path, index=False, header=False)
            warnings.warn(f"Found {num_misisng} centroids not present in the graph!\nFull list available in {output_path}\n") + str(centroids[~present_centroids].head()))
```